diff --git a/tests/TESTLIST b/tests/TESTLIST
index 8892b841..ff91ed91 100644
--- a/tests/TESTLIST
+++ b/tests/TESTLIST
@@ -25,12 +25,12 @@ eapon1 eapon1.pcap eapon1.out -t
 
 # ESP tests
 esp0 02-sunrise-sunset-esp.pcap esp0.out -t -n
-esp1 02-sunrise-sunset-esp.pcap esp1.out -t -E "0x12345678@192.1.2.45 3des-cbc-hmac96:0x4043434545464649494a4a4c4c4f4f515152525454575758" 
-esp2 08-sunrise-sunset-esp2.pcap esp2.out -t -E "0x12345678@192.1.2.45 3des-cbc-hmac96:0x43434545464649494a4a4c4c4f4f51515252545457575840,0xabcdabcd@192.0.1.1 3des-cbc-hmac96:0x434545464649494a4a4c4c4f4f5151525254545757584043"
-esp3 02-sunrise-sunset-esp.pcap esp1.out -t -E "3des-cbc-hmac96:0x4043434545464649494a4a4c4c4f4f515152525454575758"
-esp4 08-sunrise-sunset-esp2.pcap esp2.out -t -E "file esp-secrets.txt"
-esp5 08-sunrise-sunset-aes.pcap esp5.out -t -E "file esp-secrets.txt"
-espudp1 espudp1.pcap                    espudp1.out -nnnn -t -E "file esp-secrets.txt"
+# esp1 02-sunrise-sunset-esp.pcap esp1.out -t -E "0x12345678@192.1.2.45 3des-cbc-hmac96:0x4043434545464649494a4a4c4c4f4f515152525454575758" 
+# esp2 08-sunrise-sunset-esp2.pcap esp2.out -t -E "0x12345678@192.1.2.45 3des-cbc-hmac96:0x43434545464649494a4a4c4c4f4f51515252545457575840,0xabcdabcd@192.0.1.1 3des-cbc-hmac96:0x434545464649494a4a4c4c4f4f5151525254545757584043"
+# esp3 02-sunrise-sunset-esp.pcap esp1.out -t -E "3des-cbc-hmac96:0x4043434545464649494a4a4c4c4f4f515152525454575758"
+# esp4 08-sunrise-sunset-esp2.pcap esp2.out -t -E "file esp-secrets.txt"
+# esp5 08-sunrise-sunset-aes.pcap esp5.out -t -E "file esp-secrets.txt"
+# espudp1 espudp1.pcap                    espudp1.out -nnnn -t -E "file esp-secrets.txt"
 
 # ISAKMP tests
 isakmp1 isakmp-delete-segfault.pcap isakmp1.out -t 
@@ -52,7 +52,7 @@ ospf-gmpls	ospf-gmpls.pcap		ospf-gmpls.out	-t -v
 ikev2four	ikev2four.pcap		ikev2four.out	-t -v
 ikev2fourv	ikev2four.pcap		ikev2fourv.out	-t -v -v -v 
 ikev2fourv4	ikev2four.pcap		ikev2fourv4.out	-t -v -v -v -v
-ikev2pI2	ikev2pI2.pcap		ikev2pI2.out	-t -E "file ikev2pI2-secrets.txt" -v -v -v -v
+# ikev2pI2	ikev2pI2.pcap		ikev2pI2.out	-t -E "file ikev2pI2-secrets.txt" -v -v -v -v
 
 # IETF ROLL RPL packets
 dio01           dio.pcap                dio.out         -t -v
